src/main/java/com/stripe/param/CustomerCreateParams.java

```diff
@@ -0,0 +1,611 @@
+      this.metadata.putAll(map);
+      return this;
+    }
+
+    /**
+     * Add a key/value pair to `metadata` map. A map is initialized for the first `put/putAll` call,
+     * and subsequent calls add additional key/value pairs to the original map. See {@link
+     * CustomerCreateParams#metadata} for the field documentation.
+     */
+    public Builder putMetadata(String key, String value) {
+      if (this.metadata == null) {
+        this.metadata = new HashMap<>();
+      }
+      this.metadata.put(key, value);
+      return this;
+    }
+
+    /**
+     * An integer amount in %s that represents the account balance for your customer. Account
+     * balances only affect invoices. A negative amount represents a credit that decreases the
+     * amount due on an invoice; a positive amount increases the amount due on an invoice.
+     */
+    public Builder setAccountBalance(Long accountBalance) {
+      this.accountBalance = accountBalance;
+      return this;
+    }
+
+    public Builder setCoupon(String coupon) {
+      this.coupon = coupon;
+      return this;
+    }
+
+    public Builder setCustomer(String customer) {
+      this.customer = customer;
+      return this;
+    }
+
+    /**
+     * An arbitrary string that you can attach to a customer object. It is displayed alongside the
+     * customer in the dashboard.
+     */
+    public Builder setDescription(String description) {
+      this.description = description;
+      return this;
+    }
+
+    /**
+     * Customer's email address. It's displayed alongside the customer in your dashboard and can be
+     * useful for searching and tracking. This may be up to *512 characters*.
+     */
+    public Builder setEmail(String email) {
+      this.email = email;
+      return this;
+    }
+
+    /** A custom ID to use for the customer. */
+    public Builder setId(String id) {
+      this.id = id;
+      return this;
+    }
+
+    /**
+     * The prefix for the customer used to generate unique invoice numbers. Must be 3–12 uppercase
+     * letters or numbers.
+     */
+    public Builder setInvoicePrefix(String invoicePrefix) {
+      this.invoicePrefix = invoicePrefix;
+      return this;
+    }
+
+    /** Default invoice settings for this customer. */
+    public Builder setInvoiceSettings(InvoiceSettings invoiceSettings) {
+      this.invoiceSettings = invoiceSettings;
+      return this;
+    }
+
+    /** The customer's shipping information. Appears on invoices emailed to this customer. */
+    public Builder setShipping(Empty shipping) {
+      this.shipping = shipping;
+      return this;
+    }
+
+    /** The customer's shipping information. Appears on invoices emailed to this customer. */
+    public Builder setShipping(Shipping shipping) {
+      this.shipping = shipping;
+      return this;
+    }
+
+    public Builder setSource(String source) {
+      this.source = source;
+      return this;
+    }
+
+    public Builder setSource(Map<String, Object> source) {
```

Comment:
I don't think we should support a map here. In theory, we should go all the way and support passing the correct list of params that are valid for the `source` hash like card's number, cvc, etc.
If we do, there's a big documentation risk as we need to make it clear no one should use this.

Reply:
Ah we don't have this encoded in OpenAPI spec for now.. Let me see if I can fix it upstream..

Reply:
Yeah this is one of those where I'm ambivalent. I want us to support it properly. But then more and more merchants will send raw PANs because it looks easy in the library. Not sure what the right product decision is here. Maybe the hash is easier because less people would find it.

Reply:
As discussed, it's a now a String